Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -39,17 +39,26 @@ function Get-AzureUri
 function Get-ProxyUri
 {
     param([String] [Parameter(Mandatory=$true)] $serverUrl)
+    
+    $proxyUri = (Get-Item env:AGENT_PROXYURL).value
+    Write-Verbose -Verbose ("Reading proxy from .proxy file. Proxy url specified={0}" -f $proxyUri)
+    $proxyCreds = [pscredential]$null
 
-    $proxyUri = [Uri]$null
-    $proxy = [System.Net.WebRequest]::GetSystemWebProxy()
-
-    if ($proxy)
+    if($proxyUri)
     {
-        $proxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
-        $proxyUri = $proxy.GetProxy("$serverUrl")
+        return [System.Uri]$proxyUri
+    }
+    else
+    {
+        $proxy = [System.Net.WebRequest]::GetSystemWebProxy()
+        if ($proxy)
+        {
+            $proxy.Credentials = [System.Net.CredentialCache]::DefaultCredentials
+            $proxyUri = $proxy.GetProxy("$serverUrl")
+            Write-Verbose -Verbose ("Reading proxy from IE. Proxy url specified={0}" -f $proxyUri.OriginalString)
+            return $proxyUri
+        }  
     }
-
-    return $proxyUri
 }
```

Comment:
$proxyUri will never be null. Even if there is no proxy, $proxyUri will be equal to $serverUrl. GetSystemWebProxy.GetProxy() handles this.